libraries/AP_Baro/AP_Baro_MS5611.cpp

```diff
@@ -111,8 +111,8 @@ bool AP_Baro_MS56XX::_init()
     case BARO_MS5611:
         prom_read_ok = _read_prom_5611(prom);
         break;
-    case BARO_MS5837:
-        name = "MS5837";
+    case BARO_MS5837_30BA:
+        name = "MS5837-30BA26";
```

Comment:
```suggestion
    case BARO_MS5837:
        name = "MS5837";
```
At this point the variant is not yet known, so it can trigger on the "general" MS5837 type, before switching out the type to the variant-specific one once it has determined what that is.